modules/dataquery/static/schema.yml

```diff
@@ -0,0 +1,331 @@
+  /queries/{QueryID}/count:      
+    parameters:
+    - name: QueryID
+      in: path
+      description: the QueryID returned by posting to /queries
+      required: true
+      style: simple
+      schema:
+        type: integer  
+    get:
+      responses:
+        '200':
+          description: |-
+           A count of the number of candidate matches that would be returned if the query were to be run by the current user right now.
+           
+           This endpoint does *not* result in a new query run being generated or run the query, it only returns the count of how many candidates would match if the query *were* to be run.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  count:
+                    type: integer
+                    description: The number of candidates that match the filters 
+                    example: 34
+  /queries/{QueryID}/run/{QueryRunID}: 
+    description: |-
+      Returns the cached results of a previously run query
+      
+      Note: This endpoint is aspirational.
+    parameters:
+    - name: QueryID
+      in: path
+      description: the QueryID returned by posting to /queries
+      required: true
+      style: simple
+      schema:
+        type: integer
+    - name: QueryRunID
+      in: path
+      description: the identifier of a previous run for this QueryID.
+      required: true
+      style: simple
+      schema:
+        type: integer 
+    get:
+      responses:
+        '500':
+          description: Not implemented
+          content:
+            application/json:
+              schema:
+                type: object
+            
+components:
+  schemas:
+    AllQueries:
+      type: object
+      properties:
+        recent:
+          type: array
+          items:
+            $ref: '#/components/schemas/QueryRun'
+        shared:
+          type: array
+          items:
+            $ref: '#/components/schemas/Query'
+        topqueries:
+          type: array
+          items:
+            $ref: '#/components/schemas/Query'
+
+    Query:
+      type: object
+      properties:
+        type:
+          type: string
+          enum: ['candidates']
+          example: "candidates"
+        Name:
+          type: string
+          example: "My Query"
+        SharedBy:
+          type: array
+          items:
+            type: string
+            example: "admin"
+        Starred:
+          type: boolean
+        Shared:
+          type: boolean
+        QueryID:
+          type: integer
+          example: 3
+        fields:
+          type: array
+          items:
+            $ref: '#/components/schemas/QueryField'
+        criteria:
+          $ref: '#/components/schemas/QueryCriteriaGroup'
+      required:
+        - type
+        - fields
+    QueryRun:
+      type: object
+      properties:
+        RunTime:
```

Comment:
Could this  be add to the Query component so that /queries returns a list of Query instead?

Reply:
So a query could be build then saved without ever being run. That would result in a null value for RunTime which would still hold meaning.

Reply:
So recent refers to RunTime oor Creation/Save time?

Reply:
RunTime